src/ConversationFlow/ConversationFlow.cpp

```diff
@@ -1,46 +1,30 @@
-#include <sys/socket.h>
-#include <csignal>
-
+//#include <sys/socket.h>
+//#include <csignal>
 #include "ConversationFlow.hpp"
-#include "Connection/Connection.hpp"
 #include "ConversationPipeline/ConversationPipeline.hpp"
 #include "ConversationManager/ConversationManager.hpp"
+#include "Connection/SocketConnection.hpp"
 
 namespace Proxy
 {
     Status
-    ConversationFlow::ReadAllDataFromConnection(Connection& connection) noexcept
+    ConversationFlow::ReadAllDataFromConnection(SocketConnection& connection) noexcept
     {
-        Status status;
-        int32_t receivedData;
-        uint8_t tmpBuffer [2048];
-        signal(SIGPIPE, SIG_IGN);
-
-        receivedData = recv(connection.GetSocketfd(), tmpBuffer, sizeof(tmpBuffer), 0);
-
-        if(receivedData == -1)
-        {
-            status = Status(Status::Error::BadRecievingDataFromSocket);
-            return status;
-        }
-
-        if(receivedData == 0)
-        {
-            status = Status(Status::Success::Success);
-            return status;
-        }
-
-        connection.Buffer().Insert(tmpBuffer, receivedData);
-
-        return status;
+        connection.ReadData();
     }
 
     Status
-    ConversationFlow::SendAllDataToConnection(const ByteStream& data, Connection& recipientConnection) noexcept
+    ConversationFlow::SendAllDataFromConnection(const ByteStream& data, SocketConnection& destination) noexcept
+    {
+        destination.SendData(data);
```

Comment:
check the status returned from ReadData.
this may look like return `destination.SendData(data)`